modules/artifact/src/main/java/org/mule/module/artifact/classloader/IBMMQResourceReleaser.java

```diff
@@ -382,4 +391,51 @@ private void processThreadLocalMap(Field threadLocalMapTableField, Object thread
       }
     }
   }
+
+  /**
+   * Close JMSCCThreadPoolMaster threads pending to close to avoid thread leaks after redeployment
+   */
+  private void disposeMQThreads() {
+    List<Thread> threads = ThreadUtils.getAllThreads().stream()
+        .filter(thread -> thread.getName().equals(JMSCC_THREAD_POOL_MAIN_NAME))
+        .filter(thread -> thread.getClass().getClassLoader() == driverClassLoader)
+        .collect(Collectors.toList());
+
+    if (!threads.isEmpty()) {
+      threads.forEach(thread -> killThread(thread));
+      try {
+        killIBMWorker();
+      } catch (Throwable e) {
+        LOGGER.debug("An error occurred trying to close the WorkQueueManager", e);
+      }
+    }
+  }
+
+  /**
+   * Kill any thread alive
+   *
+   * @param thread thread to kill
+   */
+  private void killThread(Thread thread) {
```

Comment:
The name of the method and the JDoc are not accurate with what it is actually doing. Change it to something like `closeWorkerThread`. It is not really killing it in the traditional sense, it is just telling it to stop gracefully.